application_handlers_test.go

```diff
@@ -854,3 +856,114 @@ func applicationEventTestHelper(t *testing.T, c echo.Context, expectedEventType
+func TestApplicationEditPausedIntegration(t *testing.T) {
+	testutils.SkipIfNotRunningIntegrationTests(t)
+
+	req := m.ApplicationEditRequest{
+		Extra:                   []byte(`{"thing": true}`),
+		AvailabilityStatus:      request.PointerToString("available"),
+		AvailabilityStatusError: request.PointerToString(""),
+	}
+
+	body, _ := json.Marshal(req)
+
+	c, rec := request.CreateTestContext(
+		http.MethodPatch,
+		"/api/sources/v3.1/applications/1",
+		bytes.NewReader(body),
+		map[string]interface{}{
+			"tenantID": int64(1),
+		},
+	)
+
+	c.SetParamNames("id")
+	c.SetParamValues("1")
+	c.Request().Header.Add("Content-Type", "application/json;charset=utf-8")
+
+	// Modify the application so that the underlying code identifies it as "paused".
+	err := dao.DB.Model(m.Application{}).Where("id = ?", 1).UpdateColumn("paused_at", time.Now()).Error
+	if err != nil {
+		t.Error(err)
+	}
+
+	badRequestApplicationEdit := ErrorHandlingContext(ApplicationEdit)
+	err = badRequestApplicationEdit(c)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Revert the changes so other tests don't have any problems.
+	err = dao.DB.Model(m.Application{}).Where("id = ?", 1).UpdateColumn("paused_at", nil).Error
+	if err != nil {
+		t.Error(err)
+	}
+
+	if rec.Code != http.StatusBadRequest {
+		t.Errorf("Wrong return code, expected %v got %v", http.StatusBadRequest, rec.Code)
+	}
+
+	want := "can't modify a paused application. Unpause it first and try again"
+	got := rec.Body.String()
+	if !strings.Contains(got, want) {
+		t.Errorf(`unexpected body returned. Want "%s" contained in what we got "%s"`, want, got)
+	}
+}
+
+// TestApplicationEditPausedUnit tests that a "bad request" response is returned when a paused application is tried to
+// be updated. Runs on unit tests.
+func TestApplicationEditPausedUnit(t *testing.T) {
+	req := m.ApplicationEditRequest{
+		Extra:                   []byte(`{"thing": true}`),
+		AvailabilityStatus:      request.PointerToString("available"),
+		AvailabilityStatusError: request.PointerToString(""),
+	}
+
+	body, _ := json.Marshal(req)
+
+	c, rec := request.CreateTestContext(
+		http.MethodPatch,
+		"/api/sources/v3.1/applications/1",
+		bytes.NewReader(body),
+		map[string]interface{}{
+			"tenantID": int64(1),
+		},
+	)
+
+	c.SetParamNames("id")
+	c.SetParamValues("1")
+	c.Request().Header.Add("Content-Type", "application/json;charset=utf-8")
+
+	// Get the specific ApplicationDao mock which simulates that the applications are paused.
+	backupDao := getApplicationDao
+	getApplicationDao = func(c echo.Context) (dao.ApplicationDao, error) {
+		return &dao.MockApplicationDao{IsPausedReturnValue: true}, nil
+	}
+
+	badRequestApplicationEdit := ErrorHandlingContext(ApplicationEdit)
+	err := badRequestApplicationEdit(c)
+
+	// Go back to the previous DAO mock.
+	getApplicationDao = backupDao
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	if rec.Code != 400 {
```

Comment:
Only because you have this different in previous test 😁  lets use here http.StatusBadRequest too.

Reply:
Nice catch, thank you!